foxglove_data_platform/client.py

```diff
@@ -2,46 +2,65 @@
 import os
 from enum import Enum
 from io import BytesIO
+import sys
 from typing import IO, Any, Dict, List, Optional, Union
 
 import arrow
 import requests
 from typing_extensions import Protocol
 
+
 try:
-    from mcap.mcap0.records import Schema as McapSchema
-    from mcap.mcap0.reader import make_reader
+    from mcap.records import Schema as McapSchema
+    from mcap.reader import make_reader
 except ModuleNotFoundError:
+    print(
+        "Unable to import modules from `mcap`. To decode MCAP data, pip install mcap",
+        file=sys.stderr,
+    )
     McapSchema = None
     make_reader = None
 
-try:
-    from mcap_ros1.decoder import Decoder as Ros1Decoder
-except ModuleNotFoundError:
-    Ros1Decoder = None
 
-try:
-    from mcap_protobuf.decoder import Decoder as ProtobufDecoder
-except ModuleNotFoundError:
-    ProtobufDecoder = None
+def import_decoders():
+    decoders = {}
+    try:
+        from mcap_ros1.decoder import Decoder as Ros1Decoder
+
+        decoders["ros1msg"] = Ros1Decoder
+    except ModuleNotFoundError:
+        print(
+            "unable to import Decoder from `mcap_ros1.decoder`. "
+            "To decode ros1 .msg encoded messages, pip install mcap-ros1-support.",
+            file=sys.stderr,
+        )
 
+    try:
+        from mcap_protobuf.decoder import Decoder as ProtobufDecoder
+
+        decoders["protobuf"] = ProtobufDecoder
+    except ModuleNotFoundError:
+        print(
+            "unable to import Decoder from `mcap_protobuf.decoder`. "
+            "To decode protobuf encoded messages, pip install mcap-protobuf-support.",
+            file=sys.stderr,
+        )
 
-def decoder_cls_for_schema(schema: Any):
-    if not McapSchema:
-        raise Exception("Mcap library not found - please install the mcap library.")
-    if isinstance(schema, McapSchema) and schema.encoding == "ros1msg":
-        if not Ros1Decoder:
-            raise Exception(
-                "Mcap ROS1 library not found. Please install the mcap-ros1-support library."
-            )
-        return Ros1Decoder
-    if isinstance(schema, McapSchema) and schema.encoding == "protobuf":
-        if not ProtobufDecoder:
-            raise Exception(
-                "Mcap protobuf library not found. Please install the mcap-protobuf-support library."
-            )
-        return ProtobufDecoder
-    raise Exception(f"No decoder class available for schema encoding {schema.encoding}")
+    try:
+        from mcap_ros2.decoder import Decoder as Ros2Decoder
+
+        decoders["ros2msg"] = Ros2Decoder
+    except ModuleNotFoundError:
+        print(
+            "unable to import Decoder from `mcap_ros2.decoder`. "
+            "To decode ros2 CDR-encoded messages, pip install mcap-ros2-support.",
```

Comment:
Is it typical for a python module that depends on having other modules to gracefully degrade like this if the other modules are not present? If I am a user of protobuf this would be noise for me that I would never need to resolve.